update: outdated tasks and APIs (#5700)
JetBrains/kotlin-web-site: docs/topics/js/js-project-setup.md

Reviewer: nit: here, we use `allProjects` but below we use `allprojects`

## Dependencies

> To declare a dependency, use the `dependencies {}` block in the `jsMain` source set in your `build.gradle(.kts)` file:

Reviewer 2: or `webMain`?

Author: I'm not sure TBH, both Kotlin Multiplatform IDE plugin and [kmp-js-wizard](https://github.com/Kotlin/kmp-js-wizard/tree/main/src) have projects with `jsMain`, should we change that?

* Test run results in the console

For running browser tests, the plugin uses [Headless Chrome](https://chromium.googlesource.com/chromium/src/+/lkgr/headless/README.md)

> by default. You can also choose another browser to run tests in by adding the corresponding entries inside the `useKarma {}`

Reviewer 2: I believe we also need to mention the experimental support of playwright, which came in 2.4.20: https://kotlinlang.org/docs/whatsnew-eap.html#new-dsl-for-browser-testing

Author: I agree, but that will be a separate task for the final 2.4.20 release

### Building executables

For building executable JavaScript artifacts through webpack, the Kotlin Multiplatform Gradle plugin contains the `jsBrowserDevelopmentWebpack`
and `jsBrowserProductionWebpack` Gradle tasks.

* `jsBrowserDevelopmentWebpack` creates development artifacts, which are larger but take little time to create.
As such, use the `jsBrowserDevelopmentWebpack` tasks during active development.
* `jsBrowserProductionWebpack` applies dead code elimination to the generated artifacts and minifies
the resulting JavaScript file, which takes more time but generates executables that are smaller. As such, use
the `jsBrowserProductionWebpack` task when preparing your project for production use.
 
 Execute either of these tasks to obtain the respective artifacts for development or production. The generated files will
 be available in `build/kotlin-webpack` unless [specified otherwise](#distribution-target-directory).

```bash
./gradlew jsBrowserProductionWebpack
```

Note that these tasks will only be available if your target is configured to generate executable files (via `binaries.executable()`).

> To generate the distribution placed into `build/dist`, run the `jsBrowserDistribution` task instead:

Reviewer: ```suggestion
To generate the distribution in the `build/dist/<targetName>/<binaryName>` directory, run the `jsBrowserDistribution` task instead:
```

Reviewer: just to align with what we have further down.

```bash
./gradlew jsBrowserDistribution
```

> This task produces the ready-to-use distribution together with the project resources.

Reviewer: ```suggestion
This task generates a ready-to-use distribution that includes the project resources.
```

Reviewer: just to keep the terms consistent with the above used one (generate - generate)

### Configuring Node.js settings

You can configure Node.js settings for each subproject or set them for the project as a whole.

For example, the default Node.js version is currently 24.16.0, but you can use a different version for a specific subproject.
Add the following lines to your subproject's `build.gradle(.kts)` file:

<tabs group="build-script">
<tab title="Kotlin" group-key="kotlin">

```kotlin
project.plugins.withType<org.jetbrains.kotlin.gradle.targets.js.nodejs.NodeJsPlugin> {

>     project.the<org.jetbrains.kotlin.gradle.targets.js.nodejs.NodeJsEnvSpec>().version = "26.2.0"

Reviewer: is there a reason to change to this version specifically? 🤔 it feels arbitrary without any description. WDYT about adding the default intstead if we want to have a specific version (24.16.0)

Author: There was a lot of feedback regarding changing the version, so I think it makes sense to show it (since the newest versions are far from default), but let's highligh that it's just an example

Reviewer: ah I see 👍 nice, LGTM
